docs/docs/explanations/starlark-idempotent-runs.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Starlark idempotent runs
+sidebar_label: Starlark idempotent runs
+---
+
+Background
+----------
+
+Did you notice those lines in the `kurtosis run` command output?
+```console
+SKIPPED - This instruction has already been run in this enclave
+```
+To reduce execution time, Kurtosis tries to optimize each run of a Starlark package based on what has already been run 
+in the enclave. 
+
+For example if you try to run twice the exact same package, Kurtosis will skip all the instructions for the second run
```

Comment:
```suggestion
This means when you try to run the exact same package twice in a row, Kurtosis will skip all the instructions for the second run because they were already executed in the first run.
```